### PR TITLE
Enable more mir-opt tests in debug builds

### PR DESCRIPTION
**Pull request by saethlin**
This is https://github.com/rust-lang/rust/pull/121531 but for the mir-opt test suite.

### CODE REVIEW

File: tests/mir-opt/funky_arms.rs [hunk 1]
```diff
@@ -1,6 +1,5 @@
 // skip-filecheck
 // EMIT_MIR_FOR_EACH_PANIC_STRATEGY
-//@ compile-flags: --crate-type lib -Cdebug-assertions=no
```
Comment (RalfJung) [inline tests/mir-opt/funky_arms.rs:3]: Why did you remove `--crate-type lib`?

Comment (saethlin) [reply tests/mir-opt/funky_arms.rs:3]: mir-opt tests are `--crate-type=rlib` by default since https://github.com/rust-lang/rust/pull/120060. codegen tests should probably get the same treatment; it's a much better default.
